components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -2665,13 +2665,13 @@ public static APIStore getExternalAPIStore(String apiStoreName, int tenantId) th
      * @return a Map of tier names and Tier objects - possibly empty
      * @throws APIManagementException if an error occurs when loading tiers from the registry
      */
-    public static Map<String, Tier> getTiers(int tierType, String tenantDomain) throws APIManagementException {
+    public static Map<String, Tier> getTiers(int tierType, String organization) throws APIManagementException {
 
         boolean isTenantFlowStarted = false;
         try {
             PrivilegedCarbonContext.startTenantFlow();
             isTenantFlowStarted = true;
-            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(organization, true);
             int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
```

Comment:
this won't work for choreo

Reply:
I think u don't hv to start a tenant flow here. remove them. to get the tenantId, use getInternalOrganizationId(organization); method

Reply:
fixed in https://github.com/wso2/carbon-apimgt/pull/10553/commits/ac2ae3789c1ef92290e91e89f7a74571a7962869